submariner/IPV6-datapath.md

```diff
@@ -0,0 +1,190 @@
+# Submariner Enhancement for IPV6 datapath
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+IPv4, IPV6 and dual-stack networking is supported for Kubernetes cluster starting in 1.21.
+IPV6 networking allowing assignment of IPv6 addresses.
+Dual-stack networking allowing the simultaneous assignment of both IPv4 and IPv6 addresses.
+
+IPv4/IPv6 dual-stack on your Kubernetes cluster provides the following features:
+
+* Dual-stack Pod networking (a single IPv4 and IPv6 address assignment per Pod)  
+* IPv4 and IPv6 enabled Services  
+* Pod off-cluster egress routing (eg. the Internet) via both IPv4 and IPv6 interfaces
+
+Currently, Submariner only supports IPV4 datapath and this proposal explains the changes required to support IPV6 and dual-stack clusters.
+
+## Overview
+
+Submariner’s control plane uses a central broker component to facilitate the exchange of metadata information between participating clusters.
+
+Both inter-cluster datapath and service-discovery (Lighthouse) apply their configuration based on the information received from control plane.
+
+The Broker, inter-cluster datapath and Lighthouse should be updated to support mixed clusters environment (e.g: dual-stack, V6 only , V4 only).
+
+### Control plane
+
+For mixed cluster environment, we may experience connectivity issue.
+In some cases, some of the clusters and the broker may not connect.
+Therefore, it is necessary to select a broker cluster with which all the clusters can communicate.
+
+### Inter-cluster datapath
+
+The inter-cluster datapath needs to support clusters with different networking configuration as described in the next table:
+
+| clusterA networking | clusterB networking | Supported connectivity type |
+| :---- | :---- | :---- |
+| V4 | V4 | V4 |
+| V4 | V6 | N/A |
+| V6 | dual-stack | V6 |
+| V6 | V6 | V6 |
+| V4 | dual-stack | V4 |
+| dual-stack | dual-stack | V4,V6 |
+
+Verification of cluster connectivity should be added when cluster joins a clusterset.
+A cluster will only be added to clusterset if it can communicate with all existing clusters in the set.
+
+For example:
+
+V4 cluster A should successfully join the clusterset.  
+Dual-stack cluster B should successfully join the clusterset.  
+V6 cluster C should fail joining the clusterset (because it can't connect to cluster A).
```

Comment:
In the table above, its mentioned that if we have a v6 cluster and a dual-stack cluster, v6 would be the supported connectivity. Whereas this line mentions that cluster C will be unable to join the clusterSet. Is this validation done at the time of joining the cluster to the Broker?

Reply:
Yep, the validation should be done when the cluster joins the clusterset.
Cluster C (V6) won't be able to join the clusterset , because it can't connect cluster A (V4)